core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -586,7 +595,7 @@ private static void checkTieredLocality() {
    * @throws IllegalStateException if invalid configuration is encountered
    */
   public static void validate() {
-    for (Map.Entry<String, String> entry : toMap().entrySet()) {
+    for (Map.Entry<String, String> entry : Collections.unmodifiableMap(PROPERTIES).entrySet()) {
```

Comment:
does the `unmodifiableMap` do anything?

Reply:
Turn the value to resolved one.
Like WORKER_TIERED_STORE_LEVEL0_DIRS_QUOTA PriorityKey has the default value "${alluxio.worker.memory.size}", unmodifiableMap will turn it to its resolved value.

Reply:
```java
/**
     * Returns an unmodifiable view of the specified map.  This method
     * allows modules to provide users with "read-only" access to internal
     * maps.  Query operations on the returned map "read through"
     * to the specified map, and attempts to modify the returned
     * map, whether direct or via its collection views, result in an
     * <tt>UnsupportedOperationException</tt>.<p>
     *
     * The returned map will be serializable if the specified map
     * is serializable.
     *
     * @param <K> the class of the map keys
     * @param <V> the class of the map values
     * @param  m the map for which an unmodifiable view is to be returned.
     * @return an unmodifiable view of the specified map.
     */
    public static <K,V> Map<K,V> unmodifiableMap(Map<? extends K, ? extends V> m) {
        return new UnmodifiableMap<>(m);
    }
```

Why does this resolve values?

Reply:
The Collections.unmodifiableSet(Set<? extends T>) will create wrapper on the original set. This wrapper set can not be modified. but still the original set can be modified.

I did not find the reasons why Collections.unmodifiedMap help resolves values, but the result is that it resolves. I guess this may happen because the original map contains the default value directly (like a string ${alluxio.worker.memory.size}, the Collections.unmodifiedMap create a wrapper, when you get values from that wrapper, the JVM will directly resolve the values?

Reply:
The JVM doesn't know anything about our convention of substituting `${property}` with the value for the property. If the original map contains the raw string "${alluxio.worker.memory.size}", the unmodifiable map will return that raw string as well. If the unmodifiable map is returning a resolved string, the `Properties` map must also contain the resolved string. We shouldn't merge code like this where we don't understand why it works

Reply:
Agreed! We should not rely on code that we do not understand. I chose another way to resolve nested properties. PTAL, thanks!

Reply:
I find that this is because Configuration.init() actually not init all the things. The set ignored properties is still exist in the Configuration and cause all the Configuration.validate() in @after function to fail. Could I stick to Collections.unmodifiableMap(PROPERTIES) here and after I figure out the root reason, I create another PR?